### PR TITLE
Onboarding functional tests

### PR DESCRIPTION
**Pull request by MattLichtenstein**
## Description

- Functional tests for the new onboarding experience
- Moved old "onboarding" test to `testInitialize.js`

## Reference

N/A

### CODE REVIEW

File: tests/functional/testOnboarding.js [hunk 1]
```diff
@@ -5,79 +5,269 @@
...
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.ONBOARDING_VIEW, 'currentIndex'), 1);
+    assert.equal(await vpn.getSetting('onboardingStep'), 1);
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.PRIVACY_BLOCK_ADS_CHECKBOX, 'checked'), 'false');
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.PRIVACY_BLOCK_TRACKERS_CHECKBOX, 'checked'), 'false');
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.PRIVACY_BLOCK_MALWARE_CHECKBOX, 'checked'), 'false');
+    assert.equal(await vpn.getSetting('dnsProviderFlags'), 0);
+
+    //Clicks block ads checkbox
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.PRIVACY_BLOCK_ADS_CHECKBOX.visible());
+    // await vpn.wait(2000)
```
Comment (Gela) [inline tests/functional/testOnboarding.js:61]: Remove this?

File: tests/functional/testOnboarding.js [hunk 2]
```diff
@@ -5,79 +5,269 @@
...
+    await vpn.waitForQuery(queries.screenOnboarding.ONBOARDING_VIEW.visible());
+
+    //Ensure we are on the first onboarding step
+    await vpn.waitForQuery(queries.screenOnboarding.DATA_SLIDE.visible());
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.ONBOARDING_VIEW, 'currentIndex'), 0);
+    assert.equal(await vpn.getSetting('onboardingStep'), 0);
+
+    //Quit and relaunch the app
+    await vpn.quit()
+    await setup.startAndConnect()
```
Comment (Gela) [inline tests/functional/testOnboarding.js:227]:
```suggestion
    await vpn.quit();
    await setup.startAndConnect();
```

File: tests/functional/testOnboarding.js [hunk 3]
```diff
@@ -5,79 +5,269 @@
...
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.ONBOARDING_VIEW, 'currentIndex'), 0);
+    assert.equal(await vpn.getSetting('onboardingStep'), 0);
+
+    //Proceed to second onboarding step
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DATA_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+
+    //Quit and relaunch the app
+    await vpn.quit()
+    await setup.startAndConnect()
```
Comment (Gela) [inline tests/functional/testOnboarding.js:240]:
```suggestion
    await vpn.quit();
    await setup.startAndConnect();
```

File: tests/functional/testOnboarding.js [hunk 4]
```diff
@@ -5,79 +5,269 @@
...
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.ONBOARDING_VIEW, 'currentIndex'), 1);
+    assert.equal(await vpn.getSetting('onboardingStep'), 1);
+
+    //Proceed to third onboarding step
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.PRIVACY_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+
+    //Quit and relaunch the app
+    await vpn.quit()
+    await setup.startAndConnect()
```
Comment (Gela) [inline tests/functional/testOnboarding.js:253]:
```suggestion
    await vpn.quit();
    await setup.startAndConnect();
```

File: tests/functional/testOnboarding.js [hunk 5]
```diff
@@ -5,79 +5,269 @@
...
+    assert.equal(await vpn.getQueryProperty(queries.screenOnboarding.ONBOARDING_VIEW, 'currentIndex'), 2);
+    assert.equal(await vpn.getSetting('onboardingStep'), 2);
+
+    //Proceed to fourth/final onboarding step
+    await vpn.waitForQueryAndClick(queries.screenOnboarding.DEVICES_NEXT_BUTTON.visible());
+    await vpn.waitForQuery(queries.screenOnboarding.STEP_NAV_STACK_VIEW.ready());
+
+    //Quit and relaunch the app
+    await vpn.quit()
+    await setup.startAndConnect()
```
Comment (Gela) [inline tests/functional/testOnboarding.js:266]:
```suggestion
    await vpn.quit();
    await setup.startAndConnect();
```

**Review by Gela — APPROVED** [review]
This is even more comprehensive than I'd hoped, thank you so much!

File: tests/functional/testOnboarding.js (hunk 3 above)
Comment (lesleyjanenorton) [reply tests/functional/testOnboarding.js:240]: ❤️

**Pull request merged**